web/webapi.py

```diff
@@ -499,9 +506,11 @@ def setcookie(
         morsel["domain"] = domain
     if secure:
         morsel["secure"] = secure
-    value = morsel.OutputString()
     if httponly:
-        value += "; httponly"
+        morsel["httponly"] = True
+    value = morsel.OutputString()
+    if samesite and samesite.lower() in ["strict", "lax"]:
```

Comment:
Use a tuple instead of a list because it is slightly more memory efficient and can not be modified at runtime.